### PR TITLE
Email and Phone nudge #982

### PR DESCRIPTION
**Pull request by billyhunt**
[Link to ZenHub issue.](https://github.com/usdoj-crt/crt-portal-management/issues/982)

## What does this change?

We have added a modal window if a user does not fill out contact information.  

**NOTE**:  We might want to consider either changing the language on the modal OR only showing the modal if a user does not fill out a phone OR an email.  Currently, if either are missing, the modal pops, which might be confusing.

## Screenshots (for front-end PR):

image

### Author

+ [x] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [ ] Check for [accessibility](/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [x] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](/docs/a11y_plan.md).
+ [x] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/cts_forms/templates/forms/report_base.html [hunk 1]
```diff
@@ -104,6 +105,7 @@ <h2 class="h1">{{ current_step_title }}</h2>
     </div>
   </div>
 {% include "partials/unsupported-browser-modal.html" %}
+{% include "partials/contact-info-confirmation-modal.html" %}
```
Comment (LindsayYoung) [inline crt_portal/cts_forms/templates/forms/report_base.html:108]: This is also in report_contact_info.html, and I think that is the better place for it.

File: crt_portal/cts_forms/templates/forms/report_base.html [hunk 2]
```diff
@@ -122,16 +124,18 @@ <h2 class="h1">{{ current_step_title }}</h2>
...
       wordLimitReachedText: "{{ word_count_text.wordLimitReachedText }}"
     };
+    root.CRT.stageNumber = {{ stage_number }}
   })(window)
 </script>
 <script src="{% static 'js/word_count.js' %}"></script>
 <script src="{% static 'js/modal.js' %}"></script>
 <script src="{% static 'js/unsupported_browsers.js' %}"></script>
-<script nonce="{{request.csp_nonce}}">
+<script src="{% static 'js/contact_info_confirmation_modal.js' %}"></script>
```
Comment (LindsayYoung) [inline crt_portal/cts_forms/templates/forms/report_base.html:133]: In the future, we might want to add an extra block for the JS that is only on one page, but I am not too worried about it for this PR.

File: crt_portal/cts_forms/templates/forms/report_base.html [hunk 3]
```diff
@@ -122,16 +124,18 @@ <h2 class="h1">{{ current_step_title }}</h2>
...
+    <script nonce="{{request.csp_nonce}}">
   (function(root) {
     if (root.CRT.isUnsupportedBrowser()) {
-      var modal = document.getElementById('unsupported_browser_modal');
-      root.CRT.openModal(modal);
+      var unsupportedBrowserModal = document.getElementById('unsupported_browser_modal');
```
Comment (LindsayYoung) [inline crt_portal/cts_forms/templates/forms/report_base.html:137]: I like descriptive names!

**Review by LindsayYoung — COMMENTED** [review]
Minor questions: 

When there is focus on the "no" option, the text turns red. Is that intentional? I couldn't find the popup in invision for some reason.

We don't need to be pixel perfect, but it looks like in the screenshot there is a bit more space between the text body and the option buttons.

File: crt_portal/cts_forms/templates/forms/report_base.html (hunk 1 above)
Comment (billyhunt) [reply crt_portal/cts_forms/templates/forms/report_base.html:108]: That makes sense!

File: crt_portal/cts_forms/templates/forms/report_contact_info.html [hunk 4]
```diff
@@ -25,3 +27,20 @@
     {% include "forms/question_cards/single_form.html" with field=form.servicemember ally_id="servicemember-help-text" %}
   </div>
 {% endblock %}
+
+{%  block submit-button %}
+    <br />
+    <a class="usa-button button--continue report-step-1-continue">
+        Next
```
Comment (LindsayYoung) [inline crt_portal/cts_forms/templates/forms/report_contact_info.html:34]: I think this needs {% trans 'Next' %}

Comment (LindsayYoung) [inline crt_portal/cts_forms/templates/forms/report_contact_info.html:34]:
```suggestion
        {% trans 'Next' %}
```

**Review by LindsayYoung — APPROVED** [review]
I found one little missing translation tag. I can approve this pr and if you feel comfortable you can merge after accepting the small change.

**Pull request merged**